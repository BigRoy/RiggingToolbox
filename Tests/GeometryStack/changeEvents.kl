
require RiggingToolbox;

inline __compile_Vec3Attribute_copyFrom() { Vec3 a[]; Vec3 b[]; Vec3Attribute_copyFrom<<<1@true>>>(a, b); }
inline __compile_pushModifier_pushPositionsAlongNormals() { Vec3 a[]; Vec3 b[]; Scalar c; pushModifier_pushPositionsAlongNormals<<<1@true>>>(a, b, c); }

operator entry(){

  StartFabricProfiling();

  GeometryStack stack();
  PolygonMeshSphereGenerator sphereGenerator(2.0, 1, true, true);
  PushModifier pushModifier(3.0);
  stack.addGeometryOperator(sphereGenerator);
  stack.addGeometryOperator(pushModifier);

  EvalContext context();
  stack.evaluate(context);
  pushModifier.setPushDist(6.0);
  stack.evaluate(context);

  StopFabricProfiling();

  report( GetEvalPathReport() );
  report( stack.getDesc());
}

