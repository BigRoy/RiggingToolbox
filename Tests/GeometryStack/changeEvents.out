function GeometrySet Ref<GeometryStack>.evaluate!(EvalContext)
--function Ref<PolygonMeshSphereGenerator>#evaluate!(EvalContext, io GeometrySet)
--function Ref<PushModifier>#evaluate!(EvalContext, io GeometrySet)
function Ref<GeometryStack>#notify!(Notifier, String, String):PushModifier.changed
--function Ref<GeometrySet>.rewind!(UInt32):1
function GeometrySet Ref<GeometryStack>.evaluate!(EvalContext)
--function Ref<PushModifier>#evaluate!(EvalContext, io GeometrySet)

GeometryStack { 
  geomOperators:[ 
    {
      type: PolygonMeshSphereGenerator
    }
    { 
      type: PushModifier
    }
  ],
  geomSet:   { 
    geometries:[ 
      Sphere:{
      },
    ]
    attributeGenerations:{ 
    },
  }
}
